src/minio/managers/policy_manager.py

```diff
@@ -91,25 +98,100 @@ def _parse_list_output(self, stdout: str) -> List[str]:
+                target_type,
+                target_name,
+                lambda current: self.add_path_access_to_policy(
+                    current, path, permission_level
+                ),
+            )
+
+    async def remove_path_access_for_target(
+        self,
+        target_type: PolicyTarget,
+        target_name: str,
+        path: str,
+    ) -> None:
         """
-        async with self.operation_context("update_policy"):
-            await self._update_minio_policy(policy_model)
-            logger.info(f"Updated policy: {policy_model.policy_name}")
-            return policy_model
+        Revoke path access from a target (user or group) using a safe read-modify-write.
+
+        The method acquires the distributed lock for the target's policy, reloads the
+        latest policy while holding the lock, applies the change (remove path access),
+        and persists via the shadow-policy update flow.
+
+        Args:
+            target_type: The policy target type (user or group).
+            target_name: The username or group name.
+            path: The S3 path to revoke (e.g., "s3a://bucket/data/project/").
+        """
+        async with self.operation_context("remove_path_access_for_target"):
+            await self._update_policy_for_target_with_transform(
+                target_type,
+                target_name,
+                lambda current: self.remove_path_access_from_policy(current, path),
+            )
+
+    async def _load_policy_for_target(
+        self, target_type: PolicyTarget, target_name: str
+    ) -> tuple[str, PolicyModel]:
+        """
+        Resolve policy name for a target and load its current PolicyModel.
+        """
+        if target_type == PolicyTarget.USER:
+            policy_name = self.get_policy_name(PolicyType.USER_HOME, target_name)
+            policy_model = await self.get_user_home_policy(target_name)
+        elif target_type == PolicyTarget.GROUP:
+            policy_name = self.get_policy_name(PolicyType.GROUP_HOME, target_name)
+            policy_model = await self.get_group_policy(target_name)
+        else:
+            raise PolicyOperationError(
+                f"Unsupported target type for path access update: {target_type}"
+            )
+        return policy_name, policy_model
+
+    async def _update_policy_for_target_with_transform(
+        self,
+        target_type: PolicyTarget,
+        target_name: str,
+        transform: Callable[[PolicyModel], PolicyModel],
+    ) -> None:
+        """
+        This helper ensures consistency by acquiring the distributed lock for the
+        resolved policy, reloading the latest policy document while holding the lock,
+        applying the provided transformation function to produce a new PolicyModel,
+        and then persisting the change using the shadow-policy flow.
+        """
+        policy_name, _ = await self._load_policy_for_target(target_type, target_name)
```

Comment:
Is this loading the policy from Minio just to get the name? I thought there was a method that does that without a load

Reply:
Yeah, looks like self.get_policy_name would do it

Reply:
I separated `_load_policy_for_target` in to two methods - `_get_policy_name_for_target` and `_load_policy_for_target`.